src/libs/ReportActionsUtils.js

```diff
@@ -76,7 +97,55 @@ function isConsecutiveActionMadeByPreviousActor(reportActions, actionIndex) {
     return currentAction.action.actorEmail === previousAction.action.actorEmail;
 }
 
+/**
+ * Get the message text for the last action that was not deleted
+ * @param {Number} reportID
+ * @param {Object} [actionsToMerge]
+ * @return {String}
+ */
+function getLastVisibleMessageText(reportID, actionsToMerge = {}) {
+    const parser = new ExpensiMark();
+    const actions = _.toArray(lodashMerge({}, allReportActions[reportID], actionsToMerge));
+    const sortedActions = _.sortBy(actions, 'sequenceNumber');
+    const lastMessageIndex = _.findLastIndex(sortedActions, action => (
+        !isDeletedAction(action)
+    ));
+    const htmlText = lodashGet(actions, [lastMessageIndex, 'message', 0, 'html'], '');
+    const messageText = parser.htmlToText(htmlText);
+    return ReportUtils.formatReportLastMessageText(messageText);
+}
+
+/**
+ * @param {Number} reportID
+ * @param {Object} [actionsToMerge]
+ * @param {Number} deletedSequenceNumber
+ * @param {Number} lastReadSequenceNumber
+ * @return {String}
+ */
+function getNewLastReadSequenceNumberForDeletedAction(reportID, actionsToMerge = {}, deletedSequenceNumber, lastReadSequenceNumber) {
+    // If the action we are deleting is unread then just return the current last read sequence number
+    if (deletedSequenceNumber > lastReadSequenceNumber) {
+        return lastReadSequenceNumber;
+    }
+
+    // Otherwise, we must find the first previous index of an action that is not deleted and less than the lastReadSequenceNumber
+    const actions = _.toArray(lodashMerge({}, allReportActions[reportID], actionsToMerge));
+    const sortedActions = _.sortBy(actions, 'sequenceNumber');
+    const lastMessageIndex = _.findLastIndex(sortedActions, action => (
+        !isDeletedAction(action) && action.sequenceNumber <= lastReadSequenceNumber
+    ));
+
+    // It's possible we won't find any and in that case the last read should be reset
+    if (lastMessageIndex < 0) {
+        return 0;
+    }
+
+    return actions[lastMessageIndex].sequenceNumber;
```

Comment:
At this point, is the code sure that `actions[lastMessageIndex]` will always exist? It kind of looks like that case is covered in the `return 0`. Is that right?